sdk/core/Azure.Core/tests/TestFramework/TestEnvironment.Recorded.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Identity;
+
+namespace Azure.Core.Testing
+{
+    public partial class TestEnvironment
+    {
+        private TestRecording _recording;
+        private bool _playback;
+
+        public void SetRecording(TestRecording recording, bool playback)
+        {
+            _credential = null;
+            _recording = recording;
+            _playback = playback;
+        }
+
+        partial void GetRecordedValue(string name, ref string value)
+        {
+            if (_recording == null)
+            {
+                return;
+            }
+
+            value =  _recording.GetVariable(name, null);
+        }
+
+        partial void GetIsPlayback(ref bool playback)
```

Comment:
Why is it partial? Will we have different implementations for different builds?

Reply:
Some test projects (EventHubs/ServiceBus) don't reference the HTTP/recording/playback part of test framework so I have to split TestEnvironment in two. This will go away when I turn test framework into a project instead of shared source.